BookingProject/BookingWebAPI/BookingWebApi.Infrastructure/Data/AppUserRepository.cs

```diff
@@ -31,5 +35,79 @@ public async Task<bool> UserExists(string userId)
             if (user == null) return false;
             return true;
         }
+
+        public async Task<Result<UserChangeDto>> Update(string userId, UserUpdateQuery query)
+        {
+            var user = await _context.Users.FindAsync(userId);
+
+            if(user is null)
+            {
+                return Result<UserChangeDto>.Failure("User wasn`t found");
+            }
+
+            var changes = new List<UserChange>();
+
+            if(!String.IsNullOrEmpty(query.Email))
+            {
+                changes.Add(new UserChange
+                {
+                    FieldName = "Email",
+                    OldValue = user.Email,
+                    NewValue = query.Email
+                });
+                user.Email = query.Email;
+            }
+            if (!String.IsNullOrEmpty(query.UserName))
+            {
+                changes.Add(new UserChange
+                {
+                    FieldName = "UserName",
+                    OldValue = user.UserName,
+                    NewValue = query.UserName
+                });
+                user.UserName = query.UserName;
+            }
+
+            if (query.CustomUserData != null)
+            {
+                var existingData = !string.IsNullOrEmpty(user.CustomUserData)
+                    ? JsonSerializer.Deserialize<List<UserCustomData>>(user.CustomUserData)
+                    : new List<UserCustomData>();
+
+                var newData = query.CustomUserData;
+
+                foreach (var kvp in newData)
+                {
+                    existingData.Add(kvp);
+                }
+
+                changes.Add(new UserChange
+                {
+                    FieldName = "Custom Data",
+                    OldValue = user.CustomUserData,
+                    NewValue = JsonSerializer.Serialize(existingData)
+                });
+
+                user.CustomUserData = JsonSerializer.Serialize(existingData);
+            }
```

Comment:
Remove code duplication. Better move it to a separate service and use the service here. BC ur method grows really fast

Reply:
Remove code duplication. Better move it to a separate service and use the service here. BC ur method grows really fast